### ISSUE TITLE
sync: fetch blocks that cross positive threshold in tortoise

### ISSUE THREAD

**Issue by dshulyak** [spacemeshos/go-spacemesh#3609]
more info: https://community.spacemesh.io/t/deferred-block-fetching-a-k-a-dangling-pointers/309
depends on: https://github.com/spacemeshos/go-spacemesh/issues/3607 https://github.com/spacemeshos/go-spacemesh/issues/3608

- [ ] implementation plan

**Comment by dshulyak** [comment]
see https://github.com/spacemeshos/go-spacemesh/issues/3608#issuecomment-1515859095 for tortoise interface. beside that expected changes in sync should be minimal. syncer should not eagerly ask peers for block. there should be two conditions to ask for a block:
- peer shared a valid certificate within hdist
- tortoise outputs that block is in consensus, but missing data locally
